pkg/global/apply_configuration.go

```diff
@@ -24,33 +25,48 @@ import (
 // to all Buildbarn binaries, regardless of their purpose.
 func ApplyConfiguration(configuration *pb.Configuration) error {
 	// Push traces to Jaeger.
-	if jaegerConfiguration := configuration.GetJaeger(); jaegerConfiguration != nil {
-		pe, err := prometheus_exporter.NewExporter(prometheus_exporter.Options{
-			Registry:  prometheus.DefaultRegisterer.(*prometheus.Registry),
-			Namespace: "bb_storage",
-		})
-		if err != nil {
-			return util.StatusWrap(err, "Failed to create the Prometheus stats exporter")
-		}
-		view.RegisterExporter(pe)
-		if err := view.Register(ocgrpc.DefaultServerViews...); err != nil {
-			return util.StatusWrap(err, "Failed to register ocgrpc server views")
+	if tracingConfiguration := configuration.GetTracing(); tracingConfiguration != nil {
+		if jaegerConfiguration := tracingConfiguration.GetJaeger(); jaegerConfiguration != nil {
+			pe, err := prometheus_exporter.NewExporter(prometheus_exporter.Options{
+				Registry:  prometheus.DefaultRegisterer.(*prometheus.Registry),
+				Namespace: "bb_storage",
+			})
+			if err != nil {
+				return util.StatusWrap(err, "Failed to create the Prometheus stats exporter")
+			}
+			view.RegisterExporter(pe)
+			if err := view.Register(ocgrpc.DefaultServerViews...); err != nil {
+				return util.StatusWrap(err, "Failed to register ocgrpc server views")
+			}
+			zpages.Handle(nil, "/debug")
```

Comment:
Is this code specific to Jaeger, by the way? What does it even do? I'm pretty much an idiot when it comes to OpenTracing.

Reply:
It's not actually, broke this out into a few different settings.  Basically, hooking up an exporter from opencensus's span stats and such to prometheus, registering stuff from grpc as needing to be tracked (it's... complex?), and hooking up a debug web page.